auth/presentation/src/test/java/com/sergius/auth/presentation/LoginViewModelTest.kt

```diff
@@ -71,4 +76,48 @@ class LoginViewModelTest {
             assertThat(awaitItem()).passwordNotVisible()
         }
     }
+
+    @Test
+    fun `test login successful`() = runTest {
+        authRepository.setLoginResult(Result.Success(Unit))
+
+        viewModel.events.test {
+            viewModel.onAction(SignInScreenAction.OnLoginClick)
+            assertThat(awaitItem()).loginSuccessful()
+        }
+    }
+
+    @ParameterizedTest
+    @MethodSource("eventErrorData")
+    fun `test login returns network error`(error: DataError.Network, resourceId: Int) = runTest {
+        authRepository.setLoginResult(Result.Error(error))
+        val expectedResourceId = UiText.StringResource(resourceId).id
+
+        viewModel.events.test {
+            viewModel.onAction(SignInScreenAction.OnLoginClick)
+            assertThat(awaitItem()).hasExpectedErrorResourceId(expectedResourceId)
+        }
```

Comment:
I don't quite get this code - you're putting the `resourceId` in the `StringResource` just to extract back the id 🤔 Am I misunderstanding something, or is this equivalent to doing something like `listOf("Hello")[0]`?